bait-files/src/main/java/com/google/LoginFilter.java

```diff
@@ -0,0 +1,88 @@
+package com.google;
+
+import static com.google.sps.data.Keys.ID_ENTITY_PROPERTY;
+import static com.google.sps.data.Keys.USER_ENTITY;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.net.URLEncoder;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.PrintWriter;
+
+
+@WebFilter("/")
+public class LoginFilter implements Filter {
+  @Override
+  public void init(FilterConfig config) throws ServletException {
+    // If you have any <init-param> in web.xml, then you could get them
+    // here by config.getInitParameter("name") and assign it as field.
```

Comment:
I'd get rid of this comment that sounds like it came from a web tutorial =)

Reply:
Removed